src/System.Slices/src/System/MemoryUtils.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    static class MemoryUtils
+    {
+        [DllImport("msvcrt.dll", CallingConvention = CallingConvention.Cdecl)]
+        private static extern unsafe int memcmp(void* first, void* second, long sizeInBytes);
+
```

Comment:
Not sure if it worth it dealing with PInvokes here and msvcrt could be not portable enough.
I think there are ways to implement pretty decent memcmp entirely managed. There are some examples around BCL. Like String.EqualsHelper.